.circleci/config.yml

```diff
@@ -73,7 +73,7 @@ workflows:
               only: /^v([0-9])+.([0-9])+.([0-9])+.*/
       - cross-compile:
           requires:
-            - build
+            - setup-and-lint
```

Comment:
@owais 

I think this PR is ready to be merged, however, should I make the same change here as you made in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/384/files and replace build with cross-compile? (I had changed this dependency to speed up the build since this PR adds another step that's dependeny on cross-compile)